### PR TITLE
Spec: auctionBuyerKeys reporting

### PR DESCRIPTION
**Pull request by alexmturner**
Adds spec handling for the per-buyer latency and statistics reporting for Protected Audience, see
https://github.com/WICG/turtledove/blob/main/FLEDGE_extended_PA_reporting.md. Some follow-up work remains and has been tracked in issues added.

***
<a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/private-aggregation-api/pull/117.html" title="Last updated on Feb 22, 2024, 9:43 PM UTC (11329b8)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/private-aggregation-api/117/23d4eb4...11329b8.html" title="Last updated on Feb 22, 2024, 9:43 PM UTC (11329b8)">Diff</a>

### CODE REVIEW

**Comment by alexmturner** [comment]
Hey @qingxinwu, could you PTAL? Tried to put together a first go at the spec for `auctionReportBuyers` -- I haven't handled the seller capabilities part just yet, but added some issues. (Also added a couple of issues for things I spotted while specing.) Thanks!

cc @caraitto

**Comment by qingxinwu** [comment]
> Hey @qingxinwu, could you PTAL? Tried to put together a first go at the spec for `auctionReportBuyers` -- I haven't handled the seller capabilities part just yet, but added some issues. (Also added a couple of issues for things I spotted while specing.) Thanks!
> 
> cc @caraitto

note there's a PR to add seller capabilities to Protected Audience's spec, and I just took over the PR and am working on it, so hopefully you'll not need to worry about it in your spec.

File: spec.bs
Comment (qingxinwu) [inline spec.bs:1532]: it seems these two steps are identical to check "If |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"]
    [=map/exists=] and |auctionConfig|'s <a spec="turtledove" for="auction config">interest group buyers</a> is not null:".

But the implementation seems not checking this, but just simply copying {{AuctionAdConfig/auctionReportBuyerKeys}}

File: spec.bs
Comment (qingxinwu) [inline spec.bs:1540]: [the implementation](https://crsrc.org/c/third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=738f1250512a5e6ff8cacb6b36540cd50e8156d8;l=2067) seems not doing this, but just simply copying all keys.
So there's a difference in implementation that the step above will continue throw errors if the condition meets for extra |index|.

File: spec.bs
Comment (qingxinwu) [inline spec.bs:1550]: [=map/For each=] should work.

File: spec.bs
Comment (qingxinwu) [inline spec.bs:1565]: seems to be this todo in code? https://crsrc.org/c/third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=738f1250512a5e6ff8cacb6b36540cd50e8156d8;l=2065

But agreed that this needs to be validated.

File: spec.bs
Comment (qingxinwu) [inline spec.bs:1574]: [=list/For each=]. But as commented above, we can leave this to Protected Audience spec.

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Removed this whole section (given that requiredSellerCapabilities is being handled in your PR)

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Ack, removed.

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Done!

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Done

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Done

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Removed

File: spec.bs
Comment (alexmturner) [reply spec.bs:1567]: Yep, seems to be the same concept

File: spec.bs
Comment (alexmturner) [reply spec.bs:1539]: Switched this to a continue and added a note to match the validation logic. And yeah I thought it might be easier to follow in the spec if we didn't just keep these as a list. We could consider switching to more closely match the implementation structure.

**Comment by alexmturner** [comment]
Thanks for the review! Should be ready for another pass :)

**Review by qingxinwu — COMMENTED** [review]
generally looks good now

**Review by qingxinwu — COMMENTED** [review]
generally looks good now

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Added a link! I'm not sure I'll be able to review the PR before next week, so maybe let's leave caleb as the main reviewer. But I'll try and take a look when I get a chance :)

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Changed

File: spec.bs
Comment (alexmturner) [reply spec.bs]: Added link

File: spec.bs
Comment (alexmturner) [reply spec.bs:1866]: Yep, totally agree -- I tried to keep the scope of this smaller to unblock the debug mode change, but I've added an issue (number 37) to more formally spec the values here.

**Comment by alexmturner** [comment]
Thanks for the review!

**Review by qingxinwu — APPROVED** [review]

**Pull request merged**
